pkg/plugins/golang/v3/commons.go

```diff
@@ -95,21 +95,41 @@ manifests: controller-gen`
 		}
 
 		if err := util.ReplaceInFile("go.mod",
-			"k8s.io/api v0.22.1",
+			"k8s.io/api v0.23.0",
 			"k8s.io/api v0.21.2"); err != nil {
 			log.Warnf("unable to update the go.mod with k8s.io/api v0.21.2: %s", err)
 		}
 
 		if err := util.ReplaceInFile("go.mod",
-			"k8s.io/apimachinery v0.22.1",
+			"k8s.io/apimachinery v0.23.0",
 			"k8s.io/apimachinery v0.21.2"); err != nil {
 			log.Warnf("unable to update the go.mod with k8s.io/apimachinery v0.21.2: %s", err)
 		}
 
 		if err := util.ReplaceInFile("go.mod",
-			"k8s.io/apimachinery v0.22.1",
-			"k8s.io/apimachinery v0.21.2"); err != nil {
-			log.Warnf("unable to update the go.mod with k8s.io/apimachinery v0.21.2: %s", err)
+			"k8s.io/client-go v0.23.0",
+			"k8s.io/client-go v0.21.2"); err != nil {
+			log.Warnf("unable to update the go.mod with k8s.io/client-go v0.21.2: %s", err)
+		}
+
+		// During the scaffolding phase, this gets added to go.mod file, running go mod tidy bumps back
+		// the version from 21.2 to the latest
```

Comment:
Sort of a hack, I can open another PR to may be fix it in the scaffolding logic ... but that seems beyond the scope of this PR imo.

Reply:
The rollback in the scaffolds is required to generate the scaffold when/if the user try to use a CRD v1beta1. Then we try to do the rollback to ensure the backwards compatibilities. From 1.22 is no longer possible to work with v1beta1 CRDs,